pymongo/auth_aws.py

```diff
@@ -88,7 +96,13 @@ def _authenticate_aws(credentials, sock_info):
                 # SASL complete.
                 break
     except PyMongoAuthAwsError as exc:
+        # Clear the cached credentials if we hit a failure in auth.
+        set_cached_credentials(None)
         # Convert to OperationFailure and include pymongo-auth-aws version.
         raise OperationFailure(
             "%s (pymongo-auth-aws version %s)" % (exc, pymongo_auth_aws.__version__)
         )
+    except BaseException:
```

Comment:
BaseException or Exception? Do we care about things like KeyboardInterrupt here?

Reply:
Exception seems reasonable, updated